File: maximum-ice-cream-bars.ts
{
    function maxIceCream(costs: number[], coins: number): number {
        let c = 0;

        costs.sort((a,b) => a - b)
        
        for (const cost of costs) {
            coins -= cost

            if (coins < 0) 
                return c

            c++
        }
        
        return c
    };

    console.log(maxIceCream([1,3,2,4,1], 7))
}

File: maximum-ice-cream-bars.rs
pub fn max_ice_cream(mut costs: Vec<i32>, coins: i32) -> i32 {
    costs.sort();
    costs
        .iter()
        .scan(0, |mut cum_sum, &cost| {
            *cum_sum += cost;
            if *cum_sum <= coins {
                Some(*cum_sum)
            } else {
                None
            }
        })
        .count() as i32
}
